lib/sqlalchemy/dialects/postgresql/psycopg2.py

```diff
@@ -766,11 +867,22 @@ def on_connect(conn):
             return None
 
     def do_executemany(self, cursor, statement, parameters, context=None):
-        if self.psycopg2_batch_mode:
-            extras = self._psycopg2_extras()
-            extras.execute_batch(cursor, statement, parameters)
-        else:
+        if self.psycopg2_executemany_mode is EXECUTEMANY_DEFAULT:
             cursor.executemany(statement, parameters)
+        elif self.psycopg2_executemany_mode is EXECUTEMANY_VALUES and \
+                context and \
+                context.compiled.execute_values_insert_template:
+
+            self._psycopg2_extras().execute_values(
+                cursor,
+                statement,
+                parameters,
+                template=context.compiled.execute_values_insert_template,
+                page_size=self.psycopg2_executemany_page_size)
+
+        else:  # EXECUTEMANY_VALUES of non-insert query, or EXECUTEMANY_BATCH
+            self._psycopg2_extras().execute_batch(cursor, statement,
```

Comment:
**mike bayer** (zzzeek) wrote:

why aren't we using pagesize here?

Reply:
We can, I didn't want to change the behavior for people not using the new flag.
If you want I'll add it